### PR TITLE
feat: RP server - use vc service to verify vc

### LINKED ISSUE #33: RP server - use vc service to verify vc

### PR DESCRIPTION
**Pull request by fqutishat**
closes #33

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: pkg/restapi/rp/operation/operations.go [hunk 1]
```diff
@@ -17,32 +31,139 @@ type Handler interface {
...
+func (c *Operation) verifyVC(w http.ResponseWriter, r *http.Request) {
+	if err := r.ParseForm(); err != nil {
+		c.writeErrorResponse(w, http.StatusInternalServerError,
+			fmt.Sprintf("failed to parse form: %s", err.Error()))
+
+		return
+	}
+
+	respData, err := c.httpPost(
+		c.vcsURL+"/verify", httpContentTypeJSON, []byte(r.Form.Get("vcDataInput")))
```
Comment (troyronda) [inline pkg/restapi/rp/operation/operations.go]: suggest defining constants for paths.

File: pkg/restapi/rp/operation/operations.go [hunk 2]
```diff
@@ -17,32 +31,139 @@ type Handler interface {
...
+	}
+}
+
+func (c *Operation) httpPost(url, commContentType string, data []byte) (string, error) {
+	resp, err := c.client.Post(url, commContentType, bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+
+	if resp != nil {
```
Comment (troyronda) [inline pkg/restapi/rp/operation/operations.go]: should do the resp == nil case in the block.

File: pkg/restapi/rp/operation/operations.go (hunk 1 above)
Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go]: done

File: pkg/restapi/rp/operation/operations.go (hunk 2 above)
Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go]: https://github.com/trustbloc/edge-sandbox/pull/35/files#diff-141aceae11185429c1572fa974caf361R150

File: pkg/restapi/rp/operation/operations.go [hunk 3]
```diff
@@ -17,32 +32,139 @@ type Handler interface {
...
+		if resp.StatusCode != http.StatusOK {
+			return "", fmt.Errorf("received unsuccessful POST HTTP status from "+
+				"[%s, %v]", url, resp.StatusCode)
+		}
+		// handle response
```
Comment (sandrask) [inline pkg/restapi/rp/operation/operations.go]: comment in wrong spot

File: pkg/restapi/rp/operation/operations.go [hunk 4]
```diff
@@ -17,32 +32,139 @@ type Handler interface {
...
+	}
+}
+
+func (c *Operation) httpPost(url, commContentType string, data []byte) (string, error) {
+	resp, err := c.client.Post(url, commContentType, bytes.NewBuffer(data))
+	if err != nil {
+		return "", err
+	}
+
+	if resp != nil {
```
Comment (sandrask) [inline pkg/restapi/rp/operation/operations.go]: Can resp ever be nil?

File: pkg/restapi/rp/operation/operations.go (hunk 3 above)
Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go]: done

File: pkg/restapi/rp/operation/operations.go (hunk 4 above)
Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go]: remove check

**Pull request merged**